ora_migrator--0.9.2.sql

```diff
@@ -711,6 +809,12 @@ BEGIN
     */
    s := regexp_replace(s, '"([^"]*)"\."([^"]*)"\."nextval"',
                           'nextval(''"\1"."\2"'')', 'gi');
+   /*
+    * Retrieve only two first arguments from to_date()
+    * Excludes NLS options
+    */
+   s := regexp_replace(s, 'to_date\((''[^,]*''),\s?(''[^,]*'').*\)',
+                          'to_date(\1, \2)', 'i');
```

Comment:
- the regex should allow spaces before the comma too
- the final `.*` is greedy and could match more than it should
- this will fail for `to_date` calls that have expressions as arguments, but I guess that is OK

I wonder: why do you need that? The skipped parameter is `nlsparam`, which describes the language of the string. If we skip that, isn't the function call bound to fail in many cases?

Reply:
From my partition specifications in `partitioning.sql` test file, some expressions have been extended with an unwanted NLS_CALENDAR option

```sql
...
SUBPARTITION part4_a_2020 VALUES LESS THAN (TO_DATE('01-JAN-2021','dd-MON-yyyy')),
```

The `HIGH_VALUE` column from `dba_tab_(sub)partitions` looks like:
```sql
TO_DATE(' 2021-01-01 00:00:00', 'SYYYY-MM-DD HH24:MI:SS', 'NLS_CALENDAR=GREGORIAN')
```

As you suggested, I added more space cases to fit with PostgreSQL `to_date` function. https://regexr.com/77mnj

Reply:
```diff
--- a/ora_migrator--0.9.2.sql
+++ b/ora_migrator--0.9.2.sql
@@ -871,7 +871,7 @@ BEGIN
     * Retrieve only two first arguments from to_date()
     * Excludes NLS options
     */
-   s := regexp_replace(s, 'to_date\((''[^,]*''),\s?(''[^,]*'').*\)',
+   s := regexp_replace(s, 'to_date\(\s?(''[^,]*'')\s?,\s?(''[^,]*'').*\)',
                           'to_date(\1, \2)', 'i');
```

Reply:
Thanks. That is certainly good enough for the purposes of the fixed-format Oracle partition boundary specifications, but if it is used on a default expression, the last `.*` would change

```none
to_date(' 2021-01-01 00:00:00', 'SYYYY-MM-DD HH24:MI:SS', 'NLS_CALENDAR=GREGORIAN') + 3 * (1 + 2)
```

to

```none
to_date(' 2021-01-01 00:00:00', 'SYYYY-MM-DD HH24:MI:SS')
```

How about

```none
regexp_replace(
   s,
   'to_date\(\s*(''[^,]*'')\s*,\s*(''[^,]*'')[^)]*\)',
   'to_date(\1, \2)',
   'i'
)
```